Disable telemetry for known notifications

pvollan opened the pull request:
#### 00575a1a4740a524567833f23e451e4a74be366e
<pre>
Disable telemetry for known notifications
<a href="https://bugs.webkit.org/show_bug.cgi?id=274279">https://bugs.webkit.org/show_bug.cgi?id=274279</a>
<a href="https://rdar.apple.com/128213400">rdar://128213400</a>

Reviewed by Sihui Liu.

Disable telemetry for notifications that we know are being observed in the WebContent process.
For in-process notifications, we do not need to forward them from the UI process.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::registerNotificationObservers):
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/279065@main">https://commits.webkit.org/279065@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/632a68e737db6eb15efcf2c4ae0b5021a70257f4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52308 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/31640 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/4729 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/55582 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3031 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/38098 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/2730 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42544 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/1935 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54404 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/29295 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45145 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23619 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/52152 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/26523 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1190 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/48423 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/2570 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57178 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27434 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/2606 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/49941 "Found 11 new test failures: webgl/2.0.0/conformance/canvas/rapid-resizing.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb32f-rgb-float.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb565-rgb-unsigned_byte.html, webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgba16f-rgba-float.html, webgl/2.0.y/conformance2/textures/webgl_canvas/tex-3d-rgb16f-rgb-float.html, webgl/2.0.y/conformance2/textures/webgl_canvas/tex-3d-rgb16f-rgb-half_float.html, webgl/2.0.y/conformance2/textures/webgl_canvas/tex-3d-rgb32f-rgb-float.html, webgl/2.0.y/conformance2/textures/webgl_canvas/tex-3d-rgb565-rgb-unsigned_byte.html, webgl/2.0.y/conformance2/textures/webgl_canvas/tex-3d-rgb565-rgb-unsigned_short_5_6_5.html, webgl/2.0.y/conformance2/textures/webgl_canvas/tex-3d-rgba16f-rgba-float.html ... (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/28667 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45265 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49183 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11445 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/29579 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28412 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebKit/Scripts/process-entitlements.sh:
```diff
@@ -207,28 +207,33 @@ function notify_entitlements()
-    plistbuddy Add :com.apple.private.darwin-notification.introspect:17 string com.apple.system.lowpowermode
-    plistbuddy Add :com.apple.private.darwin-notification.introspect:18 string com.apple.system.timezone
-    plistbuddy Add :com.apple.private.darwin-notification.introspect:19 string com.apple.zoomwindow
+    plistbuddy Add :com.apple.private.darwin-notification.introspect:8 string com.apple.accessibility.cache.guided.access
```
This seems not in `const Vector<ASCIILiteral> notificationMessages`, is it forwarded?

pvollan committed:
Disable telemetry for known notifications
https://bugs.webkit.org/show_bug.cgi?id=274279
rdar://128213400

Reviewed by Sihui Liu.

Disable telemetry for notifications that we know are being observed in the WebContent process.
For in-process notifications, we do not need to forward them from the UI process.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::registerNotificationObservers):
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: https://commits.webkit.org/279065@main
